components/BuyAssetForm/index.tsx

```diff
@@ -52,12 +52,13 @@ const BuyAssetForm = ({
     }
 
     try {
+      const chainAccount = currentUser.actor;
       const purchaseResult = await ProtonSDK.purchaseSale({
-        buyer: currentUser ? currentUser.actor : '',
+        buyer: currentUser ? chainAccount : '',
```

Comment:
Let's keep it as it was so we don't possibly receive an `unable to access "actor" or undefined object` sort of error.

Reply:
Scratch that! We should just set an error and return early when `currentUser` is undefined.